python/graphstorm/config/argument.py

```diff
@@ -3677,16 +3699,28 @@ def _add_task_general_args(parser):
                 "the evaluation metric used. Supported metrics are accuracy,"
                 "precision_recall, or roc_auc multiple metrics"
                 "can be specified e.g. --eval-metric accuracy precision_recall")
-    group.add_argument('--report-eval-per-type', type=bool, default=argparse.SUPPRESS,
-            help="Whether report evaluation metrics per node type or edge type."
-                 "If True, report evaluation results for each node type/edge type."
-                 "If False, report an average evaluation result.")
+    group.add_argument(
+        '--report-eval-per-type', type=lambda x: (str(x).lower() in ['true', '1']),
```

Comment:
We should open another PR to fix similar codes.

Reply:
Yeah I just found this right above the new arg so added a fix.